src/psyclone/domain/common/transformations/alg_trans.py

```diff
@@ -0,0 +1,102 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2021, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author R. W. Ford STFC Daresbury Lab
+
+'''Specialise generic PSyIR representing an algorithm layer to a
+PSyclone algorithm-layer-specific PSyIR which uses specialised classes.
+
+'''
+from psyclone.psyGen import Transformation
+from psyclone.psyir.nodes import Call, Routine, Container
+from psyclone.domain.common.transformations import InvokeTrans
+from psyclone.psyir.transformations import TransformationError
+
+
+class AlgTrans(Transformation):
+    '''Transform a generic PSyIR representation of the Algorithm layer to
+    a PSyclone version with specialised domain-specific nodes.
+
+    :param str invoke_name: the name used to specify an invoke \
+        call. This is an optional argument that defaults to 'invoke'.
+
+    '''
+    def __init__(self, invoke_name="invoke"):
```

Comment:
I am curious, is it ever not "invoke"?

Does it depend on the domain? In this case it would be enough to construct it without parameter:
self._invoke_name = "invoke"
and let a Domain specialisation of this class change it if needed? I will prefer this as transformations do not generally have construct parameters.

Or are you thinking about any other different use case?

Reply:
No, it was a feature of the previous implementation and I was carrying it over. All of our domains that have a concept of an algorithm layer use the keyword "invoke", this is just allowing a different name to be used. You are quire right that it should really be a domain specialisation and as we don't even use it I will remove it in this PR.